app/main.py

```
@@ -1,2 +1,102 @@
+            raise KeyError("Missing key")
+        while self.storage[index].key != key:
+            index = (index + 1) % self.capacity
+            if self.storage[index] is None:
+                raise KeyError("Missing key")
+        return self.storage[index].value
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.need_resize():
+            self.resize()
+        hash_ = hash(key)
+        index = hash_ % self.capacity
+        node = Node(
+            key=key,
+            hash=hash_,
+            value=value
+        )
+        self.__set_item(node, index)
+
+    def __set_item(self, node: Node, index: int) -> None:
+        while True:
+            if self.storage[index] is None:
+                self.storage[index] = node
+                self.length += 1
+                break
+            if self.storage[index].key == node.key:
+                self.storage[index] = node
+                break
+            index = (index + 1) % self.capacity
+
+    def __len__(self) -> int:
+        return self.length
+
+    def need_resize(self) -> bool:
+        return self.length >= self.capacity * 2 // 3
+
+    def resize(self) -> None:
+        self.capacity *= 2
+        old_storage = self.storage
+        self.storage = [None] * self.capacity
+        self.length = 0
+        for item in old_storage:
+            if not item:
+                continue
+            index = item.hash % self.capacity
+            self.__set_item(item, index)
+
+    def clear(self) -> None:
+        self.storage = [None] * 8
+        self.capacity = 8
+        self.length = 0
+
+    def get(self, key: Any, default: Any = None) -> Any:
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
+    def pop(self, key: Any, default: Any = None) -> Any:
+        value = self.get(key, default)
+        if value == default:
+            raise KeyError("Missing key")
+        del self[key]
+        return value
```

The logic to detect if a key was found is flawed. The condition `if value == default` will incorrectly raise a `KeyError` if the key exists but its stored value happens to be the same as the `default` value (`None` by default). A more robust way to handle this is to use a unique sentinel object to check if `get` returned its default because the key was missing.

Additionally, the behavior doesn't fully align with Python's `dict.pop(key, default)`. The standard method returns the `default` value if the key is not found, without raising an error. Your version raises a `KeyError` in this case, which makes the `default` parameter's purpose misleading.